Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -1925,6 +1925,7 @@ template <class TreeBuilder> TreeStatement Parser<LexerType>::parseBlockStatemen
             break;
         case BlockType::StaticBlock:
             newScope->setIsStaticBlock();
+            newScope->setAllowsVarDeclarations();
```

Comment:
It's weird that we `preventVarDeclarations` at line 1921 above only to `setAllowsVarDeclarations` it here.  Why not only `preventVarDeclarations` for the `BlockType::CatchBlock` and `BlockType::Normal` cases?  With this, you don't need to add `setAllowsVarDeclarations`.